hachoir/parser/file_system/ext2.py

```diff
@@ -235,6 +239,16 @@ def describe_file(self):
                 return out + ' (-> %s)' % (self['link_target'].value)
         return out
 
+    def is_fast_symlink(self):
+        self.seekByte(4 * 15 + 4)
+        acl = UInt32(self, "file_acl")
```

Comment:
You may have to restore the previous file offset (self.current_size), after creating the field, no?

Reply:
Should I do like following?
```
def is_fast_symlink(self):
       tmp = self.current_size
        self.seekByte(4 * 15 + 4)
       acl = UInt32(self, "file_acl")
       self.current_size = tmp
```

Reply:
Yes, does it work?

Reply:
> Yes, does it work?

It looks for me that my original code without restoring `current_size` works fine as my test case shows.  

I tried the following code:
```
     def is_fast_symlink(self):
+        self.warning ("before: %d"% self.current_size)
         self.seekByte(4 * 15 + 4)
         acl = UInt32(self, "file_acl")
+        self.warning ("after: %d"% self.current_size)
```

What I got is:
```
[1][warn] [root|/group[0]/inode_table/inode[12]] before: 320
[2][warn] [root|/group[0]/inode_table/inode[12]] after: 320
```

So I think saving and restoring the value of `currenst_size` don't make sense.
I guess the value is updated only when the method returns the new field with `yield`.

If none knows the truth, I will read hachoir more.  However, you may know everything.

Reply:
Do I have to save/restore current_size field anyway for merging?